src/odemis/gui/util/img.py

```diff
@@ -2313,6 +2346,9 @@ def images_to_export_data(streams, view_hfw, view_pos,
 
     # Create legend for print-ready
     if not raw:  # png, tiff
+        # In print-ready export, a fake canvas is used by the ruler overlay
+        if orig_canvas.ruler_overlay and fake_canvas:
```

Comment:
I don't think you need to check for `fake_canvas` here. It's created when i == 0, so all the time.
Or if you want to be extra careful, I'd put it in a sub `if` block and in the else log a big warning to indicate something unexpected happened.